assets/js/graphql/get-traits.js

```diff
@@ -140,4 +141,59 @@ export function traitSearchFunctionFactory(...callbacks) {
         });
         return promise;
     };
-}
\ No newline at end of file
+}
+
+/**
+ * Creates a callback function that can be used with the `traitSearchFunctionFactory` function
+ * to convert `GWAS` in the `PaginatedSearchResults<TraitSearchResult[]>` into links that
+ * open a modal with the given `modalId`.
+ * @param {string} modalId - The HTML `id` of the target modal element.
+ * @returns {Function} The created callback function.
+ */
+export function GWASModalLinkFactory(modalId) {
+  return ({results: oldResults, ...pageInfo}) => {
+    const results = oldResults.map(({identifier, type, ...traitInfo}) => {
+      const data = {identifier, type: 'GWAS'};
+      return {
+        ...traitInfo,
+        identifier: (type == "GWAS" ? modalLink(modalId, identifier, data) : identifier),
+        type: type
+      }
+    });
+    return {...pageInfo, results};
+  }
+}
+
+/**
+ * Creates a callback function that can be used with the `traitSearchFunctionFactory` function
+ * to convert `QTLStudies` in the `PaginatedSearchResults<TraitSearchResult[]>` into links that
+ * open a modal with the given `modalId`.
+ * @param {string} modalId - The HTML `id` of the target modal element.
+ * @returns {Function} The created callback function.
+ */
+export function QTLStudiesModalLinkFactory(modalId) {
+  return ({results: oldResults, ...pageInfo}) => {
+    const results = oldResults.map(({identifier, type, ...traitInfo}) => {
```

Comment:
The logic here is fine but the ternary is a bit awkward. Here's an alternative:
```js
    const results = oldResults.map(({identifier: oldIdentifier, type, ...traitInfo}) => {
      let identifier = oldIdentifier;
      if (type === "QTL") {
        const data = {identifier, type: 'QTLStudy'};
        identifier = modalLink(modalId, identifier, data);
      }
      return {
        ...traitInfo,
        identifier,
        type,
      }
    });
```
Ditto for `GWASModalLinkFactory`.

Reply:
OK, I think I've address your requested changes in 97dc271; note that the renaming of allModalLinksFactory in get-genes.js is probably technically a breaking change, though I was aiming at a minor version bump. Let me know if I misunderstood your intent or if you indeed plan for me to reflect the non-backward compatibility in a major version bump.